### PR TITLE
pre-defined classes-mathematical program-hard-q9

### PR DESCRIPTION
**Pull request by SamamthaH**
Wrote out clause for GrowthRate. Waiting on double literal to provide a better error message.

### CODE REVIEW

File: src/pre_defined_classes/mathematical_program/hard/q9/GrowthRateTest.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class GrowthRateTest extends BaseTest {
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The growth rate for the country is:"),
+                new DoubleLiteral(56.25, 56.25, "math")
```
Comment (SanaBob) [inline src/pre_defined_classes/mathematical_program/hard/q9/GrowthRateTest.java:14]: add a bit of space between double integral +/- 0.1 maybe

File: src/pre_defined_classes/mathematical_program/hard/q9/GrowthRateTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        };
+    }
+
+    public void runMain() {
+        GrowthRate.main(new String[]{});
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals("56.25", getItemByName("math"));
```
Comment (SanaBob) [inline src/pre_defined_classes/mathematical_program/hard/q9/GrowthRateTest.java:24]: parseDouble and error statement

File: src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //Parsons
+    public Clause[] testSentence() {
+        return new Clause[]{
+                new StringLiteral("The growth rate for the country is:"),
+                new DoubleLiteral("math")
```
Comment (Hedgemon4) [inline src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java:15]: Variable name

File: src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import global.BaseTest;
+import global.variables.Clause;
+import global.variables.clauses.*;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class MainTest extends BaseTest {
+    //Parsons
```
Comment (Hedgemon4) [inline src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java:11]: `\\ Parsons (with distractractors)`

File: src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        };
+    }
+
+    public void runMain() {
+        GrowthRate.main(new String[]{});
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(56.25, Double.parseDouble(getItemByName("math")), 0.1, "Math is incorrect");
```
Comment (Hedgemon4) [inline src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java:25]: I would provide a more specific error message

**Review by Hedgemon4 — CHANGES_REQUESTED** [review]

File: src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+        };
+    }
+
+    public void runMain() {
+        GrowthRate.main(new String[]{});
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(56.25, Double.parseDouble(getItemByName("growthRate")), 0.1, "Math is incorrect. Refer back to the question for what the correct calculation should look like");
```
Comment (Hedgemon4) [inline src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java:25]: I would say something more along the lines of growthRate is incorrect, but I think that is a personal preference.

**Review by Hedgemon4 — APPROVED** [review]

**Review by aadiraju — DISMISSED** [review]
Looks good! @Opeyem1a can look at it and merge it in!

File: src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        };
+    }
+
+    public void runMain() {
+        GrowthRate.main(new String[]{});
+    }
+
+    @Test
+    public void mathTest() {
+        assertEquals(56.25, Double.parseDouble(getItemByName("growthRate")), 0.1, "Growth rate calculation is incorrect. Refer back to the question for what the correct calculation should look like");
```
Comment (Opeyem1a) [inline src/pre_defined_classes/mathematical_program/hard/q9/MainTest.java:25]: I think delta could be `0.01` here

**Review by Opeyem1a — CHANGES_REQUESTED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Review by Opeyem1a — APPROVED** [review]

**Pull request merged**